crates/biome_js_analyze/src/lint/suspicious/no_useless_escape_in_string.rs

```diff
@@ -174,6 +179,12 @@ fn next_useless_escape(str: &str, quote: u8) -> Option<usize> {
                         return Some(i);
                     }
                 }
+                // In template literals, $\{ is a valid escape for producing a literal ${
+                b'{' => {
+                    if !(quote == b'`' && i > 0 && str.as_bytes()[i - 1] == b'$') {
```

Comment:
A couple of questions:
- the comment mentions `\`, but the code doesn't check for it. What are you checking for here?
- why do you use `as_bytes`? Why not `it.clone().next()`?